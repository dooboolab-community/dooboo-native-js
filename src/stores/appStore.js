import { observable } from 'mobx';
import { NavigationActions } from 'react-navigation';

class ObservableListStore {
  @observable cnt: number = 0;
  @observable loading: boolean = false;
}

const observableListStore = new ObservableListStore();
export default observableListStore;
