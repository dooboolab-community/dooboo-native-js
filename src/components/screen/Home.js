/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  Button,
  View,
} from 'react-native';
import appStore from '../../stores/appStore';
import { getString } from '../../../STRINGS';
import { observer } from 'mobx-react';

@observer
export default class App extends Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      cnt: 0,
    };
  }

  componentDidMount() {
    // setTimeout(() => {
    //   appStore.loading = true;
    // }, 3000);
  }

  render() {
    // if (!appStore.loading) {
    //   return (
    //     <Text>Loading</Text>
    //   );
    // }
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          {getString('HELLO')}
        </Text>
        <Button
          title='click'
          onPress={this.onClick}
        />
        <Text style={styles.instructions}>
          {/* count: {this.state.cnt}{'\n'} */}
          appStore.cnt: {appStore.cnt}
        </Text>
      </View>
    );
  }

  onClick = () => {
    // this.setState({
    //   cnt: this.state.cnt + 1,
    // });
    appStore.cnt = appStore.cnt + 3;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
