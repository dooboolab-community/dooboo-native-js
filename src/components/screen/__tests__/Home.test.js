import 'react-native';
import * as React from 'react';
import Home from '../Home';
import appStore from '../../../stores/appStore';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('Home', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Home store={appStore}/>).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('component test', () => {
    const wrapper = shallow(
      <Home store={appStore} />,
    );

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
