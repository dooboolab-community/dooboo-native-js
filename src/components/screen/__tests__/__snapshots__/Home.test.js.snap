// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test renders as expected 1`] = `
<App
  store={
    ObservableListStore {
      "cnt": 0,
      "loading": false,
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "ObservableListStore@2.keys",
          "observers": Set {},
        },
        "name": "ObservableListStore@2",
        "target": [Circular],
        "values": Map {
          "cnt" => 0,
          "loading" => false,
        },
      },
    }
  }
/>
`;

exports[`Home renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Hello
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        click
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    count: 
    0
    

    this.props.store.cnt: 
    0
  </Text>
</View>
`;
